File: scripts/r/run_all.r
# =====================================================================
# run_all.r  --  regenerate every figure and table of
#   Monjo (2025), ApJ 981, 195, from data/ using the R pipeline.
# Usage:  Rscript run_all.r
# =====================================================================
sdir <- tryCatch(dirname(normalizePath(sub("--file=", "",
        grep("--file=", commandArgs(FALSE), value = TRUE)[1]))),
        error = function(e) ".")
message("== Supplementary four-cluster figures (Angus 2008) ==")
source(file.path(sdir, "10_figure0_four_clusters.r"))
message("== Figure 1 (five complete X-COP clusters) ==")
source(file.path(sdir, "20_figure1_complete.r"))
message("== Figure 2 + Table 2 (twelve X-COP clusters) ==")
source(file.path(sdir, "30_figure2_table2.r"))
message("== DONE. Outputs in figures/. ==")

File: scripts/python/run_all.py
"""
run_all.py -- regenerate every figure and table of
  Monjo (2025), ApJ 981, 195, from data/ using the Python pipeline.
Usage:  python run_all.py
"""
import runpy
import os

HERE = os.path.dirname(os.path.abspath(__file__))
for script in ["fig0_four_clusters.py", "fig1_complete.py", "fig2_table2.py"]:
    print(f"\n===== running {script} =====")
    runpy.run_path(os.path.join(HERE, script), run_name="__main__")
print("\n== DONE. Outputs (*_python.*) in figures/. ==")
